src/git.ts

```diff
@@ -112,17 +112,58 @@ export class Git {
     }
   }
 
-  public async cherryPick(commitShas: string[], pwd: string) {
-    const { exitCode } = await this.git(
-      "cherry-pick",
-      ["-x", ...commitShas],
-      pwd,
-    );
-    if (exitCode !== 0) {
+  public async cherryPick(
+    commitShas: string[],
+    allowPartialCherryPick: boolean,
+    pwd: string,
+  ): Promise<string[] | null> {
+    const abortCherryPickAndThrow = async (
+      commitShas: string[],
+      exitCode: number,
+    ) => {
       await this.git("cherry-pick", ["--abort"], pwd);
       throw new Error(
         `'git cherry-pick -x ${commitShas}' failed with exit code ${exitCode}`,
       );
+    };
+
+    if (!allowPartialCherryPick) {
+      const { exitCode } = await this.git(
+        "cherry-pick",
+        ["-x", ...commitShas],
+        pwd,
+      );
+
+      if (exitCode !== 0) {
+        await abortCherryPickAndThrow(commitShas, exitCode);
+      }
+
+      return null;
+    } else {
+      let uncommitedShas: string[] = [...commitShas];
+
+      // Cherry-pick commit one by one.
+      for (const sha of commitShas) {
+        const { exitCode } = await this.git("cherry-pick", ["-x", sha], pwd);
+
+        if (exitCode !== 0) {
+          if (exitCode === 1) {
```

Comment:
Based on what I read, a conflict should always return `1`, but the other way around is not true. `1` might now always indicate a proper conflict. For example, I believe an "empty" cherry-pick will also return `1`. I assumed here that `1` would equal a conflict since it's very hard to be sure that a proper conflict occured that would make sense with our resolution suggestion. There are so many different ways that a conflict can occur based on git unit test: https://github.com/git/git/blob/master/t/t3507-cherry-pick-conflict.sh. I am open to suggestion, I just don't know in what way I should parse `stdout`. Also there might be a solution using the output of `git status`, but i'm not too familiar with this.

Reply:
When there is a conflict, the stdout will contain [`hint: After resolving the conflicts, mark them with`](https://github.com/git/git/blob/0f3415f1f8478b05e64db11eb8aaa2915e48fef6/t/t3507-cherry-pick-conflict.sh#L57C2-L57C53), which we could use as an indication of a conflict. It's not so pretty, but it should work and it's unlikely that this will change anytime soon. 

However, I'm also okay with just using `exitCode === 1` for now. It's just as unlikely that the exit code changes, and it doesn't need to be perfect at this time.